### DISCUSSION TITLE
Suggestion: Improved Image Upload

### DISCUSSION THREAD

**Discussion by qbjl** [ScratchAddons/ScratchAddons#4657] [Ideas] [answered]
Image upload should also take URLs (that aren't compatible with scratch's img) and upload the contents of the url to assets.scratch.mit.edu.

For example: The url `https://img.silvxrcat.repl.co/images/emojis/4551-this.png` cannot be put into an `[img]`, but if I were to upload the url to SA, it would get the image and upload that instead.

This would be useful for people who _don't_ want to download a file from a url and instead want to upload it without hogging space on their computer.

**Comment by mxmou** [comment, marked as answer]
There is an issue for this: #2107

**Reply by qbjl** [reply]
Last post made there was over a year ago... wow
